bindings/csharp/Regorus/Compiler.cs

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+#nullable enable
+namespace Regorus
+{
+    /// <summary>
+    /// Represents a policy module with an ID and content.
+    /// </summary>
+    public struct PolicyModule
+    {
+        /// <summary>
+        /// Gets or sets the unique identifier for this policy module.
+        /// </summary>
+        public string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Rego policy content.
+        /// </summary>
+        public string Content { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the PolicyModule struct.
+        /// </summary>
+        /// <param name="id">The unique identifier for this policy module</param>
+        /// <param name="content">The Rego policy content</param>
+        public PolicyModule(string id, string content)
+        {
+            Id = id;
+            Content = content;
+        }
+    }
+
+    /// <summary>
+    /// Provides static methods for compiling policies into efficient compiled representations.
+    /// These are convenience methods that create an engine internally and perform compilation.
+    /// </summary>
+    public static unsafe class Compiler
+    {
+        /// <summary>
+        /// Compiles a policy from data and modules with a specific entry point rule.
+        /// This is a convenience function that sets up an Engine internally and calls the appropriate compilation method.
+        /// </summary>
+        /// <param name="dataJson">JSON string containing static data for policy evaluation</param>
+        /// <param name="modules">List of policy modules to compile</param>
+        /// <param name="entryPointRule">The specific rule path to evaluate (e.g., "data.policy.allow")</param>
+        /// <returns>A compiled policy that can be evaluated efficiently</returns>
+        /// <exception cref="Exception">Thrown when compilation fails</exception>
+        public static CompiledPolicy CompilePolicyWithEntrypoint(string dataJson, IEnumerable<PolicyModule> modules, string entryPointRule)
+        {
+            var dataBytes = Encoding.UTF8.GetBytes(dataJson + char.MinValue);
+            var entryPointBytes = Encoding.UTF8.GetBytes(entryPointRule + char.MinValue);
+            var modulesArray = modules.ToArray();
+
+            // Convert C# modules to native structs
+            var nativeModules = new Internal.RegorusPolicyModule[modulesArray.Length];
+            var pinnedHandles = new List<GCHandle>();
+
+            try
+            {
+                for (int i = 0; i < modulesArray.Length; i++)
+                {
+                    var idBytes = Encoding.UTF8.GetBytes(modulesArray[i].Id + char.MinValue);
```

Comment:
>char.MinValue [](http://example.com/codeflow?start=78&length=13)

what are all these for?

Reply:
These are for converting C# strings to Rust which expects UTF8 encoding. Strings must be explicitly null terminated as well.